provisioning/e2e/_provisioning_e2e.js

```diff
@@ -130,99 +133,81 @@ var X509Group = function() {
   this.transports = [ Http ];
   var id = uuid.v4();
 
-  this._deviceId = 'deleteme_provisioning_node_e2e_' + id;
   this._registrationId = 'reg-' + id;
-  this._groupId = 'group-' + id;
+  this._groupId = 'testgroup';
 
   this.initialize = function(callback) {
-    debug('creating CA cert');
-    var caConfig = [
-      '[req]',
-      'req_extensions = v3_req',
-      'distinguished_name = req_distinguished_name',
-      '[req_distinguished_name]',
-      'commonName = IoTHub Test CA',
-      'commonName_max = 64',
-      '[v3_req]',
-      'basicConstraints = critical,CA:TRUE'
-    ].join('\n');
-    pem.createCertificate({ config: caConfig, commonName: 'IoTHub Test CA' }, function (err, caCert) {
+    debug('creating device cert');
+    self._factoryCert = {
+        cert: new Buffer(process.env.IOTHUB_CA_ROOT_CERT,"base64").toString(),
+        key: new Buffer(process.env.IOTHUB_CA_ROOT_CERT_KEY,"base64").toString(),
+      };
+    var deviceCertOptions = {
+      commonName: self._registrationId,
+      serviceKey: self._factoryCert.key,
+      serviceCertificate: self._factoryCert.cert,
+      serviceKeyPassword: '123',
+      serial: Math.floor(Math.random() * 1000000000),
+      days: 1,
+      config: [
+        '[req]',
+        'req_extensions = v3_req',
+        'distinguished_name = req_distinguished_name',
+        '[req_distinguished_name]',
+        'commonName = ' + self._registrationId,
+        '[v3_req]',
+        'extendedKeyUsage = critical,clientAuth'
+      ].join('\n')
+    };
+    pem.createCertificate(deviceCertOptions, function(err, cert) {
       if (err) {
         callback(err);
       } else {
-        self._caCert = {
-          cert: caCert.certificate,
-          key: caCert.serviceKey
-        };
-        debug('creating device cert');
-        var certOptions = {
-          commonName: self._registrationId,
-          serviceKey: self._caCert.key,
-          serviceCertificate: self._caCert.cert,
-          serial: Math.floor(Math.random() * 1000000000),
-          days: 1
+        self._cert = {
+          key: cert.clientKey,
+          cert: cert.certificate
         };
-        createX509Certificate(certOptions, function(err, cert) {
-          if (err) {
-            callback(err);
-          } else {
-            self._cert = cert;
-            callback();
-          }
-        });
+        fs.writeFileSync('device-cert.pem',self._cert.cert);
```

Comment:
> fs.writeFileSync('device-cert.pem',self._cert.cert) [](start = 7, length = 52)

You're doing this why?

Reply:
because I sometimes forget to remove code i used to debug the certs

---
In reply to: [154234327](https://github.com/Azure/azure-iot-sdk-node/pull/166#discussion_r154234327) [](ancestors = 154234327)